src/components/groups/group.middlewares.ts

```diff
@@ -26,9 +27,25 @@ export const joinGroup = asyncWrapper(async (req: Request, res: Response, next:
     await Group.relatedQuery('users')
       .for(group.id)
       .relate(user.id)
+    next()
```

Comment:
There should be a return here, otherwise the redirect in line 32 will run as well.